### PR TITLE
Port to Nuvola Player 3

### PR DESCRIPTION
**Pull request by thenyeguy**
I used the old Spotify plugin to get the scraping and rewrote the plugin for Nuvola Player 3. It only handles the core functionality - buttons/media keys and metadata.

### CODE REVIEW

**Comment by jiri-janousek** [comment]
Hi @thenyeguy. Thanks for your work. I will probably take a look at it in February after I release Nuvola Player 3 beta.

File: Makefile [hunk 1]
```diff
@@ -22,15 +22,33 @@
 
 # Service integration id
 APP_ID = spotify
+# Dependencies
+DEPS = rsvg-convert
 # Default installation destination
 DEST ?= $(HOME)/.local/share/nuvolaplayer3/web_apps
+# Size of PNG app icon
+ICON_SIZE ?= 48
```
Comment (jiri-janousek) [inline Makefile]: Change [icon size in Makefile to 128](https://groups.google.com/d/msg/nuvola-player-devel/a-axnC05sf8/a8hEsBolxiIJ).

File: README.md [hunk 2]
```diff
@@ -3,7 +3,9 @@ Spotify for Nuvola Player 3
 
 Integration of Spotify into your linux desktop via
 [Nuvola Player](https://github.com/tiliado/nuvolaplayer).
- 
+
+Based on the Nuvola Player 2 Spotify plugin by Chase Colman (<chase@colman.io>).
```
Comment (jiri-janousek) [inline README.md:7]: How much is your work based on Chase's original script? If there are significant code portions from his script, you have to retain copyright from the original file.

File: integrate.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+    }
+
+    // Page is ready for magic
+    WebApp._onPageReady = function()
+    {
+        // Connect handler for signal ActionActivated
+        Nuvola.actions.connect("ActionActivated", this);
+
+        // this.connect();
```
Comment (jiri-janousek) [inline integrate.js]: Remove commented-out code.

File: metadata.json [hunk 4]
```diff
@@ -1,9 +1,10 @@
 {
     "id": "spotify",
     "name": "Spotify",
-    "maintainer_name": "Chase Colman?",
-    "maintainer_link": "https://github.com/FIXME",
-    "version_major": 2,
+    "maintainer_name": "Michael Nye",
+    "maintainer_link": "https://github.com/thenyeguy",
+    "home_url": "https://play.spotify.com/browse",
+    "version_major": 1,
```
Comment (jiri-janousek) [inline metadata.json]: Revert version back to 2.0. 1.0 has been already released.

**Comment by jiri-janousek** [comment]
I've added some in-line comments. I haven't tested whether your integration actually works because I cannot make Spotify work behind HTTP proxy.

Also, could you fill in white inner parts of the Spotify logo? I think it would look better ;-)

File: README.md (hunk 2 above)
Comment (thenyeguy) [inline README.md:7]: Good point. I only used his script as a reference to see how to get information from Spotify's API. I'm not familiar with licensing stuff at all, should the original copyright be retained?

**Comment by thenyeguy** [comment]
I fixed the issues you mentioned, thanks for the feedback. The log also does look better with white!

However, I updated Nuvola Player for the first time in a while, and Spotify no longer loads on my machine. It seems to happen even with the template integrate.js on the development tutorial. I made a [gist with the debug output](https://gist.github.com/thenyeguy/e0dc7aa126a571bef427). There is one javascript error, but I don't see the same error in the console when loading Spotify in Chrome or Firefox.

It seems like a Nuvola issue, but I can't tell for sure. I could open this as an issue against Nuvola?

File: README.md (hunk 2 above)
Comment (jiri-janousek) [inline README.md:7]: It's ok if you used his script only as a reference.

**Comment by jiri-janousek** [comment]
> However, I updated Nuvola Player for the first time in a while, and Spotify no longer loads on my machine. It seems to happen even with the template integrate.js on the development tutorial. I made a gist with the debug output. There is one javascript error, but I don't see the same error in the console when loading Spotify in Chrome or Firefox.
> 
> It seems like a Nuvola issue, but I can't tell for sure. I could open this as an issue against Nuvola?

It works fine on my machine. [Create a new bug report](https://github.com/tiliado/nuvola-app-spotify/issues/new) for that. And also check whether your issue is reproducible in WebKit2 MiniBrowser. If you use Debian/Ubuntu, it should be available in one of these locations:

```
/usr/lib/x86_64-linux-gnu/webkit2gtk-3.0/libexec/MiniBrowser
/usr/lib/x86_64-linux-gnu/webkit2gtk-4.0/libexec/MiniBrowser
/usr/lib/i386-linux-gnu/webkit2gtk-3.0/libexec/MiniBrowser
/usr/lib/i386-linux-gnu/webkit2gtk-4.0/libexec/MiniBrowser
```

I will merge your branch today/tomorrow and grant you commit access to [the main repository](https://github.com/tiliado/nuvola-app-spotify). If you haven't already done so:
1. Subscribe to the [Nuvola Player Development forum](https://groups.google.com/d/forum/nuvola-player-devel) as I post here announcements related to development.
2. Create an account at [Tiliado.eu](https://tiliado.eu) and send me your username. I'll add you to Tiliado developers group. You can then fill in your credentials in Nuvola Player preferences not to be bothered by donate bar.

**Pull request merged**